src/components/pages/MainPage.spec.js

```diff
@@ -40,13 +39,39 @@ describe("<MainPage/>", () => {
     it("Does not redirect to Setup MFA page if user is federated", async (done) => {
         authMock.getCurrentUser.mockImplementation(() => ({preferredMFA: "NONE"}))
         authMock.isFederated.mockImplementation(() => true)
+
+        const wrapper = mount(<MockAuthProvider><MainPage nav={mockNav}/></MockAuthProvider>)
+
+        process.nextTick(() => {
+            act(() => {
+                wrapper.update();
+            });
+            expect(mockNav.go).not.toBeCalledWith(Pages.SETUP_MFA);
+            done();
+        });
+    });
+
+    it("Redirects to User Not Set Up page if api returns false", async (done) => {
+        authMock.getCurrentUser.mockImplementation(() => ({
+            signInUserSession:{
+                idToken: {
+                    jwtToken: "test_token"
+                }
+            },
+            preferredMFA: "SOFTWARE_TOKEN_MFA"
+        }))
+        authMock.isFederated.mockImplementation(() => false)
```

Comment:
Shouldn't we be checking against federated users, since this is the main reason behind the ticket.